templates/test/ci/cluster-template-prow-private.yaml

```diff
@@ -30,7 +30,15 @@ spec:
     creationTimestamp: ${TIMESTAMP}
     jobName: ${JOB_NAME}
   bastionSpec:
-    azureBastion: {}
+    azureBastion:
+      name: ${CLUSTER_NAME}-azure-bastion
+      publicIP:
+        name: ${CLUSTER_NAME}-azure-bastion-pip
+      subnet:
+        cidrBlocks:
+        - ${AZURE_BASTION_SUBNET_CIDR}
+        name: AzureBastionSubnet
+        role: bastion
```

Comment:
we should define a new const SubnetRole for this https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/main/api/v1beta1/types.go#L511 and make sure no assumptions are made in the code for subnet role being either control plane or node

Then, can be here or in a follow up, we should add enum validation for subnet role to make sure it's one of the supported roles (looks like it's not being validated currently if it's letting you specify an arbitrary undefined bastion role)

Reply:
Sure, I opened up issue #2000 and I'll make a PR for that.